cmd/main.go

```diff
@@ -104,7 +110,32 @@ func main() {
 		os.Exit(1)
 	}
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	restConfig := ctrl.GetConfigOrDie()
+
+	dpaClientScheme := runtime.NewScheme()
```

Comment:
Any reason we cannot just add the OADP DPA scheme to reconciler client and fetch the enforced Backup spec at the start of NAB reconciler ? before stepping into the reconcile step functions ?

Reply:
that would do a get call on every reconcile, no?

preferred to do here to do just one call and do not add oadp scheme to reconcile client

Reply:
I think @shubham-pampattiwar  was thinking of adding it to the:

```
func init() {
	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
	utilruntime.Must(nacv1alpha1.AddToScheme(scheme))
	utilruntime.Must(velerov1.AddToScheme(scheme))
```

And then using as `Scheme` within reconcile. This way it behaves same as others. The get on that can be limited by predicate ?

Reply:
I was thinking of just adding scheme here to the controller client, and then in NAB reconcile loop we could have a task func called `refreshEnforcedBackupSpec`. In this function you only update the in-memory reconciler struct if there was a change in enforcedspec via DPA. To limit the Get calls to DPA we could use the same annotation we used for restarted nac pod, `enforced-backup-spec` to determine if refresh of  enforcedspec is needed or not. Thoughts on this ?